[tasks] Include Names for `include_tasks`

the-real-cphillips opened the pull request:
Currently, the output when running this role is confusing.
It isn't clear what exactly is being skipped for each `include_tasks` play.

This was initially confusing until I dug into the actual `tasks/main.yml`
This PR adds a `name:` parameter to each `include_tasks` within the `tasks/main.yml`

While a small nit-picky PR, It makes the output clearer and easier to explain to others, without having to dive into code.

Commit:
[tasks] Include Names for `include_tasks`

Currently, the output when running this agent is not clear what exactly
is being skipped.

This just adds `name:` to each `include_tasks` in `tasks/main.yml`
This makes the output a little more friendly.

KSerrania commented:
Hello @the-real-cphillips,

Thanks for the PR! We released a new major version of the role this Wednesday, so the task list got modified. Would you mind rebasing your PR and adding the `name:` parameter to the new and modified tasks?

the-real-cphillips commented:
> Hello @the-real-cphillips,
> 
> Thanks for the PR! We released a new major version of the role this Wednesday, so the task list got modified. Would you mind rebasing your PR and adding the `name:` parameter to the new and modified tasks?

Sure thing! I'll get on it now :)

the-real-cphillips committed:
Merge branch 'master' into name_parameter_for_includes

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
-- include_tasks: agent-linux.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent6-linux.yml
```
```suggestion
  include_tasks: agent-linux.yml
```

KSerrania replied:
@the-real-cphillips I think you forgot or misclicked on this one, the main (non-Agent 5) linux install file is named [agent-linux.yml](https://github.com/the-real-cphillips/ansible-datadog/blob/name_parameter_for_includes/tasks/agent-linux.yml) now

the-real-cphillips replied:
you are correct 🙈 
I just fixed in latest commit

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
-  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent6-linux.yml
+  when: not datadog_agent5 and ansible_os_family != "Windows"
```
```suggestion
  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
```

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
-  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent5-linux.yml
+  when: datadog_agent5 and ansible_os_family != "Windows"
```
```suggestion
  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
```

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
-- include_tasks: agent-win.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family == "Windows"
+- name: Agent6 Install Tasks (Windows)
+  include_tasks: agent6-win.yml
```
```suggestion
  include_tasks: agent-win.yml
```

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
-  when: datadog_agent_major_version|int > 5 and ansible_os_family == "Windows"
+- name: Agent6 Install Tasks (Windows)
+  include_tasks: agent6-win.yml
+  when: not datadog_agent5 and ansible_os_family == "Windows"
```
```suggestion
  when: datadog_agent_major_version|int > 5 and ansible_os_family == "Windows"
```

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
 
-- include_tasks: agent-linux.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
```
```suggestion
- name: Linux Install Tasks
```

KSerrania commented on tasks/main.yml:
```diff
@@ -1,33 +1,45 @@
 
-- include_tasks: agent5-linux.yml
-  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
```
```suggestion
- name: Linux Install Tasks (Agent 5)
```

KSerrania reviewed (changes requested):
Thanks! I'm adding some suggested changes to make the role work as expected, the merge reverted some changes that were done in the major version bump (mainly some filename changes, and datadog_agent5 got replaced by datadog_agent_major_version).

Once these are added the PR should be good to go!

the-real-cphillips committed:
Update tasks/main.yml

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

the-real-cphillips committed:
Update tasks/main.yml

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

the-real-cphillips committed:
Update tasks/main.yml

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

the-real-cphillips committed:
Update tasks/main.yml

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

the-real-cphillips committed:
Update tasks/main.yml

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

the-real-cphillips committed:
Update tasks/main.yml

Co-Authored-By: Kylian Serrania <kylian.serrania@datadoghq.com>

Commit:
[Changes] Resolving Requested Changes

- missed the `agent6-linux.yml` to `agent-linux.yml`, updated here
- ran through `yamllint` and `ansible-lint`
  - no issues except for `ansible-lint` errors for 501/602

KSerrania reviewed (approved):
Thanks! 
I'm going to merge this PR and release a new version on Monday.

As you mentioned in your commits, we have a few linting errors, but they're tackled by #236 so everything's good here!

the-real-cphillips commented:
> Thanks!
> I'm going to merge this PR and release a new version on Monday.
> 
> As you mentioned in your commits, we have a few linting errors, but they're tackled by #236 so everything's good here!

Awesome thanks!

the-real-cphillips commented:
@KSerrania probably unrelated here, but has there been any talk about including molecule for testing the role to ensure it's idempotent and works as expected? 

I plausibly would be willing to undertake such a thing if there's desired need/want for it.
